### ISSUE TITLE
Crash on clone plenum space type

### ISSUE THREAD

**Issue by macumber** [NREL/OpenStudio#2720]
Code to reproduce:

```
OpenStudio::Logger.instance.standardOutLogger.enable
OpenStudio::Logger.instance.standardOutLogger.setLogLevel(OpenStudio::Debug)

m = OpenStudio::Model::Model.new
s = OpenStudio::Model::Space.new(m)
z = OpenStudio::Model::ThermalZone.new(m)
s.setThermalZone(z)
a = OpenStudio::Model::AirLoopHVACSupplyPlenum.new(m)
a.setThermalZone(z)
st = s.spaceType.get
puts st
s.hardApplySpaceType(true)
```

**Comment by macumber** [comment]
Something is going wacky in the workspace rename object code:
```

[utilities.idf.Workspace] <-1> Renamed Object of type 'OS:SpaceType' and named '
Plenum Space Type' to 'Plenum Space Type' to avoid a name conflict upon Workspac
eObject addition.
[utilities.idf.Workspace] <-1> Unable to add cloned objects to Workspace. The va
lidity report is:
The collection is INVALID at strictness level 'Draft', because of the errors:
Object     level data error of type NameConflict      .
Error pertains to an object of type 'OS:SpaceType', named 'Plenum Space Type'.
Additional information about the error type: two objects of the same type or wit
h overlapping references also have the same name.
```

**Comment by macumber** [comment]
I could look into it but am just too swamped

**Comment by eringold** [comment]
Thanks @macumber! I owe you one. :cocktail:

**Comment by jmarrec** [comment]
Backtrace with the above ruby code and a breakpoint set on resolvePotentialNameConflicts (see [Workspace.cpp#L2169](https://github.com/NREL/OpenStudio/blob/34238e7c2c48de2dd1a5dc7877693536864d0854/openstudiocore/src/utilities/idf/Workspace.cpp#L2169))

```
  * frame #0: 0x00000001207ce512 openstudio.bundle`openstudio::detail::Workspace_Impl::resolvePotentialNameConflicts(this=0x00007fa1c9b6ec50, other=0x00007ffee1d46f90, toIgnore=size=1) at Workspace.cpp:2117
    frame #1: 0x00000001207d6011 openstudio.bundle`openstudio::detail::Workspace_Impl::addAndInsertObjects(this=0x00007fa1c9b6ec50, allObjects=size=1, foundObjectIndices=size=1, notFoundObjectIndices=size=1, equivalentObjects=size=1) at Workspace.cpp:1171
    frame #2: 0x00000001207d5094 openstudio.bundle`openstudio::detail::Workspace_Impl::addAndInsertObjects(this=0x00007fa1c9b6ec50, objectsToAdd=size=1, objectsToInsert=size=1) at Workspace.cpp:1091
    frame #3: 0x000000012080c1bc openstudio.bundle`openstudio::Workspace::addAndInsertObjects(this=0x00007ffee1d49c38, objectsToAdd=size=1, objectsToInsert=size=1) at Workspace.cpp:2937
    frame #4: 0x0000000118f8f008 openstudio.bundle`openstudio::model::detail::ModelObject_Impl::clone(this=0x00007fa1c84be6e0, model=Model @ 0x00007ffee1d4a0a8) const at ModelObject.cpp:851
    frame #5: 0x000000011971a0a2 openstudio.bundle`openstudio::model::detail::ParentObject_Impl::clone(this=0x00007fa1c84be6e0, model=Model @ 0x00007ffee1d4a1f8) const at ParentObject.cpp:117
    frame #6: 0x0000000118f95407 openstudio.bundle`openstudio::model::ModelObject::clone(this=0x00007ffee1d4a6d0, model=Model @ 0x00007ffee1d4a410) const at ModelObject.cpp:1145
    frame #7: 0x00000001199edeb7 openstudio.bundle`openstudio::model::detail::Space_Impl::hardApplySpaceType(this=0x00007fa1c9b78f50, hardSizeLoads=true) at Space.cpp:2092
    frame #8: 0x0000000119a0b182 openstudio.bundle`openstudio::model::Space::hardApplySpaceType(this=0x00007fa1bfe17160, hardSizeLoads=true) at Space.cpp:3385
    frame #9: 0x000000011bf93c97 openstudio.bundle`_wrap_Space_hardApplySpaceType(argc=1, argv=0x000000010e25f3c0, self=140332859751840) at ruby_OpenStudioModelGeometry_wrap.cxx:130

```

**Comment by jmarrec** [comment]
Fixed in #2853
